tests/external/libcxx/basic_string/string.modifiers/string_assign/iterator.pass.cpp

```diff
@@ -37,33 +189,38 @@ test(nvobj::pool<struct root> &pop, const S &s1, It first, It last,
 				[&] { r->s = nvobj::make_persistent<S>(s1); });
 
 	auto &s = *r->s;
-	// XXX: enable operator==
-	// auto &expected = *r->expected;
 
 	s.assign(first, last);
 
-	// XXX: enable operator==
-	// UT_ASSERT(s == expected);
+	UT_ASSERT(s == expected);
 
 	nvobj::transaction::run(pop,
 				[&] { nvobj::delete_persistent<S>(r->s); });
 }
 
-//#ifndef TEST_HAS_NO_EXCEPTIONS
-// template <class S, class It>
-// void
-// test_exceptions(S s, It first, It last)
-//{
-//    S aCopy = s;
-//    try {
-//        s.assign(first, last);
-//        assert(false);
-//    }
-//    catch (...) {}
-//    LIBCPP_ASSERT(s.__invariants());
-//    assert(s == aCopy);
-//}
-//#endif
+template <class S, class It>
+void
+test_exceptions(nvobj::pool<struct root> &pop, const S &s1, It first, It last)
+{
+	auto r = pop.root();
+
+	nvobj::transaction::run(pop, [&] {
+		r->s = nvobj::make_persistent<S>(s1);
+		r->aCopy = nvobj::make_persistent<S>(*r->s);
+	});
+
+	try {
+		r->s->assign(first, last);
+		UT_ASSERT(false);
+	} catch (...) {
```

Comment:
You should probably catch some specific exception.

Reply:
done